### PR TITLE
[f] Add Order management page

### PR DESCRIPTION
**Pull request by minhchau273**
- [x] [f] Add Order management page

### CODE REVIEW

File: app/views/orders/index.html.slim [hunk 1]
```diff
@@ -0,0 +1,38 @@
+.container-fluid.tickets-buy
+  .row.tickets-header
+    .col-xs-12.col-md-11.col-md-offset-1
+      h3
+        | Order Management
+      .pull-left
+        = fa_icon "ticket"
+      .venue-details
+        = "#{@orders.count} order(s)"
```
Comment (seri) [inline app/views/orders/index.html.slim]: Use Rails `pluralize` for this

File: app/views/orders/index.html.slim [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        th.center
+          | Booking Date
+        th.center
+          |  Total
+        - @orders.each_with_index do |order, index|
+          tr[data-href=order_path(order)]
+            td.center
+              = index + 1
+            td.event-name
+              = order.event_name
```
Comment (seri) [inline app/views/orders/index.html.slim]: Let's mark N+1 this too 😄

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -14,4 +14,19 @@
...
     it { is_expected.not_to allow_value("afdf", "a.com", "abs@fds.2a").for(:email) }
   end
+
+  describe "#total_amount" do
+    let(:user) { create(:user) }
+    let(:event) { create(:event) }
+    let(:ticket_type_1) { create(:ticket_type, event: event, price: 50_000) }
+    let(:ticket_type_2) { create(:ticket_type, event: event, price: 100_000) }
+    let(:order_1) { create(:order, user: user, event: event) }
+    let(:order_2) { create(:order, user: user, event: event) }
```
Comment (seri) [inline spec/models/user_spec.rb:24]: Let's create an order that doesn't belong to the current user 😄

File: spec/support/shared_examples/show_error_shared_example.rb [hunk 4]
```diff
@@ -4,3 +4,10 @@
     expect(flash[:alert]).to eq message
   end
 end
+
+shared_examples "require signing in" do
+  it "redirects to Sign in page" do
+    expect(response).to redirect_to login_path
+  end
+end
+
```
Comment (seri) [inline spec/support/shared_examples/show_error_shared_example.rb]: LOL

**Comment by minhchau273** [comment]
@seri 😓

File: app/controllers/orders_controller.rb [hunk 5]
```diff
@@ -1,10 +1,14 @@
 class OrdersController < ApplicationController
   before_action :load_event, only: :new
   before_action :check_expired_event, only: :new
-  before_action :authenticate_resource, only: [:index, :new]
+  before_action :authenticate_resource, only: [:index, :new, :create]
   before_action :load_order, only: :show
   authorize_resource only: :show
 
+  def index
+    @orders = current_user.orders
```
Comment (seri) [inline app/controllers/orders_controller.rb:9]: I see that the view is not taking care of the case when `@orders` is empty.

File: spec/controllers/orders_controller_spec.rb [hunk 6]
```diff
@@ -1,6 +1,37 @@
...
+
+      it "is successful and creates new order with some order items based on the ticket types of this event" do
+        expect(response).to be_success
+        expect(assigns(:orders)).to eq orders
+      end
+    end
+
+    context "user has not signed in yet" do
+      before do
+        expect(controller).to receive(:store_location_and_require_login).and_call_original
```
Comment (seri) [inline spec/controllers/orders_controller_spec.rb]: We shouldn't need this anymore since the logic is tested in "require signing in"

**Comment by seri** [comment]
@chauvo Just a couple things and we can merge 💯

File: app/views/orders/index.html.slim [hunk 7]
```diff
@@ -13,25 +13,29 @@
...
-              = index + 1
-            td.event-name
-              = order.event_name
-            td.center.booking-time
-              = order.created_at_to_s
-            td.center.order-total
-              = formatted_price(order.total)
+      - if @orders.empty?
+        .alert.alert-danger.order-message
+          = "You don't have any order."
```
Comment (seri) [inline app/views/orders/index.html.slim]: ?

**Pull request merged**